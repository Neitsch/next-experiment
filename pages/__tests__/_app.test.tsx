import { shallow } from "enzyme";
import React from "react";

import App from "../_app";

jest.mock("apollo-client", () => ({
  ApolloClient: ({ children }) => children,
}));
jest.mock("react-apollo", () => ({
  ApolloProvider: ({ children }) => children,
}));
jest.mock("../../lib/withApolloClient", () =>
  jest.fn().mockImplementation(x => x),
);
jest.doMock("next/app", () => {
  const Helper = ({ children }) => children;
  Helper.prototype.Container = ({ children }) => children;
  return Helper;
});

describe("App page", () => {
  it("renders", () => {
    const app = <App Component={() => <div />} apolloClient={{}} />;
    const rendered = shallow(app);
    expect(rendered).toMatchSnapshot();
  });
});
