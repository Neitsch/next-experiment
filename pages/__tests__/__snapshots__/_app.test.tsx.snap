// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page renders 1`] = `
<Container>
  <ApolloProvider
    client={Object {}}
  >
    <Component />
  </ApolloProvider>
</Container>
`;
